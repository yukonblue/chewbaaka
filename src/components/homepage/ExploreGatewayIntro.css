/**
 * ExploreGatewayIntro.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Aug 08, 2021
 */

/* div.ExploreGatewayIntro {
} */

p.JoinTheRace {
  font-size: 24px;
  font-weight: bold;
  margin: auto;
}

span.SafeTheCheetah {
  font-size: 24px;
  font-weight: bold;
  color: rgb(255,182,31);
}

span.SafeTheCheetah:hover {
  font-size: 24px;
  font-weight: bold;
  color: rgb(255,149,0);
}

div.ExploreGatewayIntroCenterGrid {
  margin: 20px auto;
}

div.ExploreGatewayIntroCenterGridLhsColumnStyle {
  display: block;
  float: right;
  max-width: 500px;
  padding-right: 20px;
  text-align: right;
}

div.ExploreGatewayIntroCenterGridRhsColumnStyle {
  max-width: 480px;
}

p.CCFMissionStatementQuote {
  display: inline;
  padding: 0.25rem 0;

  background-image: linear-gradient(45deg, rgba(255,193,7,0.75) 0%, rgba(255,193,7,0.75) 100%);
  background-repeat: no-repeat;
  background-size: 100% 0.25rem;
  background-position: 0 100%;
  
  font-family: 'Roboto', sans-serif;
  font-size: 21px;
  font-style: italic;
  font-weight: 500;
  line-height: 1.6;
  text-align: right;
}

img.ExploreGatewayIntroCCFCheetahMuseumImage {
  display: block;
  max-width: 100%;
  margin: 0 auto;
}

p.ExploreGatewayMenuIconLinkText {
  text-align: center;
}

/* a.ExploreGatewayMenuIconLink {
} */
