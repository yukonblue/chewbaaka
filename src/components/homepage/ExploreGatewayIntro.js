/**
 * ExploreGatewayIntro.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Aug 22, 2021
 */

import React from 'react'

import {
  Container,
  Header,
  Icon,
} from 'semantic-ui-react'

import 'semantic-ui-css/semantic.min.css'

import '../shared/GlobalStyles.css'

import '../shared/ContentPageSharedStyles.css'

import './ExploreGatewaySharedStyles.css'

import './ExploreGatewayIntro.css'

import image_CCF_Cheetah_Museum from './assets/CCF_Cheetah_Museum-min.jpg'

export default function ExploreGatewayIntro() {
  return (
    <div className="ExploreGatewayIntro ExploreGatewayBackground" data-testid="ExploreGatewayIntroTestId">
      <div className="VerticalCushionPadding">
        <img
          width="1000"
          height="600"
          className="ExploreGatewayIntroCCFCheetahMuseumImage"
          src={image_CCF_Cheetah_Museum}
          alt="CCF Cheetah Museum"
        />
      </div>

      <div className="VerticalCushionPadding">
        <Container width={8}>
          <Header size="huge" textAlign="center">
            Explore CCFâ€™s Cheetah Museum in Namibia and learn about the cheetahs, all at the comfort of your home.
          </Header>
          <div className="ExploreGatewayMenuIconLinkOuterContainer">
            <p className="ExploreGatewayMenuIconLinkText">
              <a href="#ExploreGatewayMenu" className="ExploreGatewayMenuIconLink">
                <Icon name='angle double down' size='huge' color='yellow'/>
              </a>
            </p>
          </div>
        </Container>
      </div>
    </div>
  );
}
