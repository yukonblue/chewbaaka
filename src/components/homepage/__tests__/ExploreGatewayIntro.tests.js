/**
 * ExploreGatewayIntro.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Aug 08, 2021
 */

import React from 'react'

import { render, screen } from '@testing-library/react'

import renderer from 'react-test-renderer'

import ExploreGatewayIntro from '../ExploreGatewayIntro'

test('renders explore gateway intro', () => {
  render(
    <ExploreGatewayIntro />
  );

  // Tests component title is present.
  const componentTitleElement = screen.getByText(/Explore CCFâ€™s Cheetah Museum/i);
  expect(componentTitleElement).toBeInTheDocument();
});

test('ExploreGatewayIntro component snapshot', () => {
  const tree = renderer
    .create(
      <ExploreGatewayIntro />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
