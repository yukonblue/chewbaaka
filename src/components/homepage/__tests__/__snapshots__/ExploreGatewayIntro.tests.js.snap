// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExploreGatewayIntro component snapshot 1`] = `
<div
  className="ExploreGatewayIntro ExploreGatewayBackground"
  data-testid="ExploreGatewayIntroTestId"
>
  <div
    className="VerticalCushionPadding"
  >
    <img
      alt="CCF Cheetah Museum"
      className="ExploreGatewayIntroCCFCheetahMuseumImage"
      height="600"
      src="CCF_Cheetah_Museum-min.jpg"
      width="1000"
    />
  </div>
  <div
    className="VerticalCushionPadding"
  >
    <div
      className="ui container"
      width={8}
    >
      <div
        className="ui huge center aligned header"
      >
        Explore CCFâ€™s Cheetah Museum in Namibia and learn about the cheetahs, all at the comfort of your home.
      </div>
      <div
        className="ExploreGatewayMenuIconLinkOuterContainer"
      >
        <p
          className="ExploreGatewayMenuIconLinkText"
        >
          <a
            className="ExploreGatewayMenuIconLink"
            href="#ExploreGatewayMenu"
          >
            <i
              aria-hidden="true"
              className="yellow angle double down huge icon"
              onClick={[Function]}
            />
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
